gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -98,18 +102,20 @@ protected void processMessage(DecodeableKafkaRecord message) {
     String key = (String) message.getKey();
     DagActionStoreChangeEvent value = (DagActionStoreChangeEvent) message.getValue();
 
-    Long timestamp = value.getChangeEventIdentifier().getTimestamp();
+    String tid = value.getChangeEventIdentifier().getTxId();
+    Long produceTimestamp = value.getChangeEventIdentifier().getProduceTimestampMillis();
     String operation = value.getChangeEventIdentifier().getOperationType().name();
     String flowGroup = value.getFlowGroup();
     String flowName = value.getFlowName();
     String flowExecutionId = value.getFlowExecutionId();
 
-    log.debug("Processing Dag Action message for flow group: {} name: {} executionId: {} timestamp {} operation {}",
-        flowGroup, flowName, flowExecutionId, timestamp, operation);
+    produceToConsumeDelayValue = calcMillisSince(produceTimestamp);
```

Comment:
as mentioned in my reply, I'm not sure that partition delay would be uniform, so, methodology-wise, to generate a singular number, it seems we presently take the last value seen during the collection period irregardless of partition.

that's questionable, from a monitoring perspective.

another option might be to use the max value during the collection period (HWM).  a variant would be some percentile, such as P99/P95 (e.g. via a t-digest).

most salient, monitoring-wise, might be the max of the per-partition most-recently-seen values.  this captures recovery/improvement during the collection window.

considering the max/HWM may be easiest to code, it seems a reasonable approximation.

Reply:
Our main concern with measuring the delays here is to flag if there's a big timing difference between the tx occurring in the store versus us being able to consume the event. I suspect the delay will more likely be on the tx occurring to brooklin producing rather than us being slow as consumers. I don't see why partition delay would actually not be uniform because if we're performing poorly overall it would be on all partitions rather than one in particular. I don't want to overcomplicate this calculation as it's just to give us a ballpark on the delay, when the more important one is the eventToProduceDelay which I will measure in another location.

Reply:
I agree on not complicating.  here's my reasoning:
* multiple incoming partitions get combined into a single input
* each partition is read into a blocking queue, so in cases of overload, the partition can't drain completely to its queue
* under such conditions, reading multiple partitions more closely resembles round-robin
* using round-robin under overload conditions, when skewed partitions, would lead those to fall farther behind than non-skewed partitions

thus, if overload and partition skew, seems possible to have non-uniform delay.

in practice, not sure how likely both those would be.  perhaps not very.

measurement-wise, against non-uniform delay, this essentially randomly chosen value would make the metric to jump around, whereas a high-water mark or percentile would consistently track worst case or each collection interval.

anyway, if my reasoning above checks out, and the conditions seem possible to arise, maybe update to the `AtomicLong`.  if they seem too unlikely, just keep as-is, but perhaps add a warning/comment about where it breaks down.  seem reasonable?